tests/test_app.py

```diff
@@ -1,59 +1,177 @@
+            headers=self.headers,
+            content_type='application/x-www-form-urlencoded',
+        )
+        if response.json['message'] != "User already exists":
+            self.assertEqual(response.status_code, 201, response.json)
 
-    def test_missing_username(self):
-        response = test_client.post('/login', headers=self.headers)
+    def test_login_missing_username(self):
+        response = test_client.post('/login', headers={ **self.headers,  'Authorization': f'Basic {self.credentials("", self.password)}'},
+)
         self.assertEqual(response.status_code, 400, response.json)
     
-    def test_unknow_username(self):
+    def test_login_unknow_username(self):
         username = str(uuid.uuid4())
-        response = test_client.post('/login', data={'username': username, 'password': 'password1'})
+        response = test_client.post(\
+            '/login',
+            headers={ **self.headers,  'Authorization': f'Basic {self.credentials(username, self.password)}'},
+            content_type='application/x-www-form-urlencoded',
+        )
         self.assertEqual(response.status_code, 401, response.json)
 
+    def test_login(self):
+        response = test_client.post(
+            '/login',
+            headers=self.headers,
+        )
+        self.assertEqual(response.status_code, 200, response.json)
+
+    def test_create_empty_inspection(self):
+        response = test_client.post('/inspections', headers=self.headers)
+        self.assertEqual(response.status_code, 500, response.json)
+
+
+    # If the inspection is created, it should at least return a message testifying that.
+    def test_create_inspection(self):
+        with requests.get('https://raw.githubusercontent.com/ai-cfia/fertiscan-pipeline/main/expected.json') as response:
+            json_data = response.json()
+            response = test_client.post(
+                '/inspections', 
+                headers=self.headers,
+                content_type='application/json',
+                json=json_data
+            )
+            self.assertEqual(response.status_code, 201, response.json)
+            self.inspection_id = response.json['inspection_id']
+
+    def test_update_empty_inspection(self):
+        inspection_id = str(uuid.uuid4())
+        response = test_client.put(f'/inspections/{inspection_id}', headers=self.headers)
+        self.assertEqual(response.status_code, 500, response.json)
+
+    # Requires the database schema
+    def test_update_inspection_fake_id(self):
+        inspection_id = str(uuid.uuid4()) # Should fail since the inspection with this id does not exist
+        response = test_client.put(
+            f'/inspections/{inspection_id}', 
+            headers=self.headers,
+            content_type='application/json',
+            json={'status': 'completed'}
+        )
+        self.assertEqual(response.status_code, 400, response.json)
+
+    # Requires the database schema
+    def test_update_inspection(self):
```

Comment:
I suggest fusing together the create inspection and the update inspection tests. That way, you can use the data returned by the first to test the second. Please keep in mind that we don't yet have a delete inspection function.

The create inspection expects  `https://raw.githubusercontent.com/ai-cfia/fertiscan-pipeline/main/expected.json` but the update inspection's model is different. Please keep the data returned by create, modify some of the fields (for example set verified=True) and use that to test update inspection.

Reply:
I stored the result of the create so it could be used in the update tests.